src/routes/AgreementCreateRoute/AgreementCreateRoute.test.js

```diff
@@ -0,0 +1,211 @@
+};
+
+const isPendingData = {
+  checkAsyncValidation: () => jest.fn(),
+  handlers: {
+    onDownloadFile: () => { },
+    onUploadFile: () => { }
+  },
+  history: {
+    push: () => jest.fn()
+  },
+  location: {},
+  mutator: {
+    agreements: {
+      POST: noop
+    },
+    query: {
+      update: noop
+    },
+  },
+  resources: {
+    loadingView,
+    agreementStatusValues,
+    basket,
+    contactRoleValues,
+    documentCategories,
+    isPerpetualValues,
+    licenseLinkStatusValues,
+    orgRoleValues,
+    query,
+    reasonForClosureValues,
+    relationshipTypeValues,
+    renewalPriorityValues,
+  }
+};
+
+describe('AgreementCreateRoute', () => {
+  describe('rendering the route with permissions', () => {
+    let renderComponent;
+    beforeEach(() => {
+      renderComponent = renderWithIntl(
+        <MemoryRouter>
+          <AgreementCreateRoute {...data} />
+        </MemoryRouter>,
+        translationsProperties
+      );
+    });
+
+    test('renders the agreementForm component', () => {
+      const { getByText } = renderComponent;
+      expect(getByText('AgreementForm')).toBeInTheDocument();
+    });
+
+    test('renders the BasketLineButton component', () => {
+      const { getByText } = renderComponent;
+      expect(getByText('BasketLineButton')).toBeInTheDocument();
+    });
+
+    // we check if the button is clicked it calls the queryUpdateMock(update) function to invoke the child callback (handleBasketLinesAdded) defined in Route
+    test('calls the BasketLineButton', async () => {
+      await ButtonInteractor('BasketLineButton').click();
+      expect(queryUpdateMock).toHaveBeenCalled();
+    });
+
+    // we check if the button is clicked it calls the historyPushMock(push) function to invoke the child callback (handleClose) defined in Route
+    test('calls the CloseButton', async () => {
+      await ButtonInteractor('CloseButton').click();
+      expect(historyPushMock).toHaveBeenCalled();
+    });
+  });
+
+  describe('rendering loading view', () => {
+    let renderComponent;
+    beforeEach(() => {
+      renderComponent = renderWithIntl(
+        <MemoryRouter>
+          <AgreementCreateRoute {...isPendingData} />
+        </MemoryRouter>,
+        translationsProperties
+      );
+    });
+
+    test('renders loadingView', () => {
+      const { getByText } = renderComponent;
+      expect(getByText('LoadingView')).toBeInTheDocument();
+    });
+  });
+
+  describe('rendering with no permissions', () => {
+    let renderComponent;
+    beforeEach(() => {
+      renderComponent = renderWithIntl(
+        <MemoryRouter>
+          <AgreementCreateRoute
+            {...data}
+            stripes={{ hasPerm: () => false }}
+          />
+        </MemoryRouter>,
+        translationsProperties
+      );
+    });
+
+    test('displays the permission error', () => {
+      const { getByText } = renderComponent;
+      expect(getByText('stripes-smart-components.permissionError')).toBeInTheDocument();
```

Comment:
Please make the changes here once the translation is added to translationProperties file